delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/directives/ResponseToJson.scala

```diff
@@ -50,12 +58,22 @@ sealed trait JsonValueInstances {
 
   implicit def uioJson(
       uio: UIO[Json]
-  )(implicit s: Scheduler, cr: RemoteContextResolution, jo: JsonKeyOrdering): ResponseToJson =
+  )(implicit
```

Comment:
I wouldn't reuse this one. This is already very complicated, let's try to avoid unneccessary extra noise.

Please provide yet another implementation (like I did with `ResponseToJson`). In your case you want `ResponseToSparqlResults` or whatever. In this case it will be more readable and won't affect this.

Reply:
I think we should keep it this way, since we need some generic way to send JSON with different `Content-Type` header. I agree that there are too many implicits but adding another `ResponseTo*` class with its own implicits conversions and magic is not really solving the problem in my opinion.

Reply:
I don't agree. `ResponseToJsonLd` deals with jsonld responses, where you can have compacted/expanded/ntriples/dot responses.

It has ways to deal with IO, UIO, plain values, etc.

In your case you don't care about all those things. You want something much more simple, that only deals with `application/sparql-query`. So basically, something very similar to `ResponseToJson`.

Please provide a solution that is scoped to what you are trying to solve, instead of broadening the sof the current `ResponseToJsonLd` even more.

Reply:
plus an implicit with default value is an anti pattern most of the times.

Reply:
There is no change in `ResponseToJsonLd`, only in `ResponseToJson`.

Reply:
Ok. In this case why don't you do the following:
1. Create an object with the following:
```scala
object ResponseToJsonType {
   def apply[E: JsonLdEncoder](mediaType: MediaType,
      uio: UIO[Either[Response[E], Complete[Json]]]
  )(implicit s: Scheduler, cr: RemoteContextResolution, jo: JsonKeyOrdering): Route =
  extractRequest { request =>
          if (HeadersUtils.matches(request.headers, mediaType)) {
            val ioRoute = uio.flatMap {
              case Left(r: Reject[E])       => UIO.pure(reject(r))
              case Left(e: Complete[E])     => e.value.toCompactedJsonLd.map(r => complete(e.status, e.headers, r.json))
              case Right(v: Complete[Json]) => UIO.pure(complete(v.status, v.headers, v.value))
            }
            onSuccess(ioRoute.runToFuture)(identity)
          } else reject(unacceptedMediaTypeRejection(mediaTypes))
        }
}
```
2. keep `ResponseToJson` as is but on line 35 just make a call to `ResponseToJsonType(`application/json`, uio)`
3. Create a `ResponseToSparql`:
```scala

sealed trait ResponseToSparql {
  def apply(): Route
}

object ResponseToSparql extends SparqlValueInstances {

  private[directives] type UseLeft[A] = Either[Response[A], Complete[SparqlResults]]
  private[directives] type UseRight   = Either[Response[Unit], Complete[SparqlResults]]

  private[directives] def apply[E: JsonLdEncoder](
      uio: UIO[Either[Response[E], Complete[SparqlResults]]]
  )(implicit s: Scheduler, cr: RemoteContextResolution, jo: JsonKeyOrdering): ResponseToSparql =
    new ResponseToSparql {

      override def apply(): Route = ResponseToJsonType(`application/sparql-query`, uio)
        
    }
}

sealed trait SparqlValueInstances {

  implicit def uioSpaqrl(
      uio: UIO[SparqlResults]
  )(implicit s: Scheduler, cr: RemoteContextResolution, jo: JsonKeyOrdering): ResponseToSparql = ???

  implicit def ioSparql[E: JsonLdEncoder: HttpResponseFields](
      io: IO[E, SparqlResults]
  )(implicit s: Scheduler, cr: RemoteContextResolution, jo: JsonKeyOrdering): ResponseToJson = ???

}

```

Reply:
In this way we don't need the extra implicit and we have the 2 cases separated, while the main block is handled in the same way inside `ResponseToJsonType`